src/main/java/net/logstash/logback/composite/FormattedTimestampJsonProvider.java

```diff
@@ -160,11 +96,11 @@ protected String getFormattedTimestamp(Event event) {
      */
     private void updateTimestampWriter() {
         if (UNIX_TIMESTAMP_AS_NUMBER.equals(pattern)) {
-            timestampWriter = new NumberTimestampWriter();
+            timestampWriter = null;
         } else if (UNIX_TIMESTAMP_AS_STRING.equals(pattern)) {
-            timestampWriter = new StringTimestampWriter();
+            timestampWriter = DateTimeFormatter.ofPattern(DEFAULT_PATTERN).withZone(TimeZone.getDefault().toZoneId());
         } else {
-            timestampWriter = new PatternTimestampWriter(FastDateFormat.getInstance(pattern, timeZone));
+            timestampWriter = DateTimeFormatter.ofPattern(pattern, Locale.getDefault()).withZone(TimeZone.getDefault().toZoneId());
```

Comment:
Since there are subtle differences in the pattern strings used by `DateTimeFormatter` and `FastDateFormat`, this is a backwards incompatible change.  I'm willing to accept this change, but this will need a major version bump of logstash-logback-encoder.

I'll probably hold off on this until logback supports timestamps with higher precision.